syntax = "proto3";

package MyVocaloid;

service Vocaloid {
  rpc Sing (SingRequest) returns (stream SingReply) {}
}

message Note {
  int32 note = 1;
  int32 start = 2;
  int32 duration = 3;
}

message SingRequest {
  int32 bpm = 1;
  string lyrics = 2;
  repeated Note notes = 3;
  int32 bank = 4;
}

message ReplyHeader {
  int32 result = 1;
  string message = 2;
}

message ReplyChunk {
  bytes data = 1;
}

message SingReply {
  oneof ReplyData {
	ReplyHeader header = 1;
	ReplyChunk chunk = 2;
  }
}

